### PR TITLE
DEPR: Replacing builtin and NumPy funcs in agg/apply/transform

### LINKED ISSUE #53425: DEPR: Special casing of NumPy and Python builtin functions
Currently various code paths detects whether a user passes e.g. Python's builtin `sum` or `np.sum` and replaces the operation with the pandas equivalent. I think we shouldn't alias a potentially valid UDF to the pandas version; this prevents users from using the NumPy version of the method if that is indeed what they want. This can lead to some surprising behavior: for example, NumPy `ddof` defaults to 0 whereas pandas defaults to 1. We also only do the switch when there are no args/kwargs provided. Thus:

    df.groupby("a").agg(np.sum)  # Uses pandas sum with a numeric_only argument
    df.groupby("a").agg(np.sum, numeric_only=True)  # Uses np.sum; raises since numeric_only is not an argument

It can also lead to different floating point behavior:

```
np.random.seed(26)
size = 100000
df = pd.DataFrame(
    {
        "a": 0,
        "b": np.random.random(size),
    }
)
gb = df.groupby("a")
print(gb.sum().iloc[0, 0])
# 50150.538337372185
print(df["b"].to_numpy().sum())
# 50150.53833737219
```

However, using the Python builtin functions on DataFrames can give some surprising behavior:

```
ser = pd.Series([1, 1, 2], name="a")
print(max(ser))
# 2

df = pd.DataFrame({'a': [1, 1, 2], 'b': [3, 4, 5]})
print(max(df))
# b
```

If we remove special casing of these builtins, then apply/agg/transform will also exhibit this behavior. In particular, whether a DataFrame is broken up into Series internally to compute the operation will determine whether we get a result like `2` or `b` above.

However, I think this is still okay to do as the change to users is straightforward: use `"max"` instead of `max`.

I've opened #53426 to see what impact this would have on our test suite.

cc @topper-123 @mroeschke @jbrockmendel

### PR DESCRIPTION
**Pull request by rhshadrach**
- [x] closes #53425 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/apply.py [hunk 1]
```diff
@@ -1695,3 +1697,23 @@ def validate_func_kwargs(
...
         raise TypeError(no_arg_message)
     return columns, func
+
+
+def warn_alias_replacement(
+    obj: AggObjType,
+    func: Callable,
+    alias: str,
+) -> None:
+    if alias.startswith("np."):
```
[the remark this answers is not in the document]
Comment (rhshadrach) [reply pandas/core/apply.py:1707]: Yes - we use lookups in `_builtin_table_alias` and `_cython_table` to identify the function the user passes, so it is independent of any binding (e.g `numpy.sum` vs `np.sum`). The `alias` here arises from the values in these two dictionaries, and so must either start with `np.` or just be the op name (e.g. "sum")

[the remark this answers is not in the document]
Comment (rhshadrach) [reply pandas/core/apply.py:1707]: Ah, but we will report that pandas is using e.g. `np.sum` regardless of how the user imports NumPy. So the warning will be emitted under the correct circumstances, but the message might not align with how they import. I think the message is still clear though.

[the remark this answers is not in the document]
Comment (mroeschke) [reply pandas/core/apply.py:1707]: Okay. Yeah I think the message is clear enough despite how the user imports numpy

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @rhshadrach

**Pull request merged**